### PR TITLE
dev/core#1410 Fix E-notice when doin a force case search with a prede…

### PR DESCRIPTION
**Pull request by seamuslee001**
…fined case subject field

Overview
----------------------------------------
This fixes an issue on the case search form when using a url like `
civicrm/case/search?reset=1&force=1&case_subject=test`

Before
----------------------------------------
Notice message `
Warning: htmlspecialchars() expects parameter 1 to be string, array given in HTML_Common->_getAttrString()` showed

After
----------------------------------------
No Notice and search completes correctly

ping @eileenmcnaughton @demeritcowboy

### CODE REVIEW

File: CRM/Case/Form/Search.php [hunk 1]
```diff
@@ -288,37 +287,6 @@ public function postProcess() {
...
   }
 
-  /**
-   * Add the rules (mainly global rules) for form.
-   *
-   * All local rules are added near the element
-   *
-   * @see valid_date
-   */
-  public function addRules() {
```
Comment (seamuslee001) [inline CRM/Case/Form/Search.php]: This was code left over after the conversion to date picker

File: CRM/Case/BAO/Query.php [hunk 2]
```diff
@@ -331,10 +333,7 @@ public static function whereClauseSingle(&$values, &$query) {
         return;
 
       case 'case_subject':
-        $query->_where[$grouping][] = CRM_Contact_BAO_Query::buildClause("civicrm_case.subject", $op, $value, 'String');
-        $query->_qill[$grouping][] = CRM_Contact_BAO_Query::getQillValue('CRM_Case_DAO_Case', $name, $value, $op, 'Case Subject');
-        $query->_tables['civicrm_case'] = $query->_whereTables['civicrm_case'] = 1;
-        $query->_tables['civicrm_case_contact'] = $query->_whereTables['civicrm_case_contact'] = 1;
+        $query->handleWhereFromMetadata($fieldSpec, $name, $value, $op);
```
Comment (seamuslee001) [inline CRM/Case/BAO/Query.php]: Figured made sense to try to standardise the thing even tho this wasn't strictly speaking the issue

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
Code looks good - @demeritcowboy do you want to test ?

**Comment by demeritcowboy** [comment]
Sure, just I get an error about patch doesn't apply because of whitespace.

**Comment by seamuslee001** [comment]
@demeritcowboy it should be against 5.20 rather than master just FYI

**Comment by demeritcowboy** [comment]
I wonder if a better solution is just `git rm CRM/Case/Form/Search.php`...

Doing both a url search `case_subject=test&force=1` and a form-based search with subject gives:
`Error: Call to undefined method CRM_Contact_BAO_Query::handleWhereFromMetadata() in CRM_Case_BAO_Query::whereClauseSingle() (line 336 of .../CRM/Case/BAO/Query.php).`

I'm also noticing something else that must have been introduced between 5.20 and master - I'll make a separate ticket.

**Comment by seamuslee001** [comment]
@demeritcowboy maybe but there is still some special handling that is in that file so i don't think we can remove it. Given your error i'm thinking maybe i need to put this against master instead or change things

@eileenmcnaughton @demeritcowboy should this be a 5.21 or 5.20 thing i can go either way

**Comment by eileenmcnaughton** [comment]
If this is about url support it's not a regression it's a new feature (out of NYSS interest) I think.

I'm all for rm -rf civicrm/*/Search/*

**Comment by demeritcowboy** [comment]
Yeah I was just joking (well, sort of). I forgot to put the smiley face.

**Comment by seamuslee001** [comment]
ok @eileenmcnaughton @demeritcowboy i have put this back to master now i think its easier that way

**Comment by demeritcowboy** [comment]
That seems good. Thanks.

**Comment by seamuslee001** [comment]
@demeritcowboy are you going to do another r-run?

**Comment by demeritcowboy** [comment]
Sorry, I wasn't clear: I did an r-run and it's good for `subject`. There are other things which are now in a separate lab ticket which aren't related to this PR.

**Comment by seamuslee001** [comment]
@demeritcowboy just opened another PR #15924 to resolve your other issue

**Comment by demeritcowboy** [comment]
Thanks!

**Comment by eileenmcnaughton** [comment]
Looks like this has been tested & code looks good - MOP

**Pull request merged**